cpp/csp/python/PyStructToJson.cpp

```diff
@@ -198,12 +198,38 @@ rapidjson::Value toJsonRecursive( const StructPtr& self, rapidjson::Document& do
     return new_dict;
 }
 
-rapidjson::Value pyDictKeyToName( PyObject * py_key, rapidjson::Document& doc )
+std::pair<const char *, Py_ssize_t> pyObjectToString( PyObject * py_obj )
 {
-    // Only support strings, ints, and floats as keys
+    auto * str_obj = PyObject_Str( py_obj );
+    Py_ssize_t len = 0;
+    const char * str = PyUnicode_AsUTF8AndSize( str_obj, &len );
```

Comment:
Note that this buffer will be deallocated when  `str_obj` is deallocated (the UTF-8 string is cached with the PyObject) so we need to copy it before decref'ing `str_obj`.

Reply:
Yeah, I completely stopped thinking about the memory leaks issue when writing this utility function, sorry about that! I was naively hoping to avoid the extra copy with this function.